packages/analytics-core/src/plugins/diagnostic.ts

```diff
@@ -0,0 +1,66 @@
+import { DiagnosticEvent, DiagnosticPlugin, Event, Result } from '@amplitude/analytics-types';
+
+export class Diagnostic implements DiagnosticPlugin {
+  name = '@amplitude/plugin-diagnostic';
+  type = 'destination' as const;
+  public serverUrl = new URL('http://localhost:8000');
```

Comment:
For configuration options see the Options pattern in other Plugins - https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/plugin-web-attribution-browser/src/typings/web-attribution.ts

Reply:
Thanks for sharing the patter. But as we not don't treat Diagnostic as a plugin, I think it would be better to have `constructor(serverUrl)` instead. Please see the latest change in `packages/analytics-core/src/diagnostic.ts`.